datalad/distribution/tests/test_publish.py

```diff
@@ -285,6 +285,10 @@ def test_publish_recursive(pristine_origin, origin_path, src_path, dst_path, sub
     # we will be publishing back to origin, so to not alter testrepo
     # we will first clone it
     origin = install(origin_path, source=pristine_origin, recursive=True)
+    # uncouple subdataset from testrepo sources after recursive install
+    # to make this clone the source of all `get` attempts
+    for sub in origin.subdatasets(result_xfm=lambda x: x['gitmodule_name']):
+        origin.subdatasets(path=sub, set_property=[('url', './{}'.format(sub))])
```

Comment:
was the intention to provide a "new" test scenario or due to code changes it became required to alter `url` entries?

Reply:
No, this is a direct consequence of https://github.com/datalad/datalad/issues/5033. The testrepos configure an explicit URL in `.gitmodules`. But here we want to test what happens if there is a relative URL (it shall clone from the initial clone target done in this test). With the change introduced by the PR we would clone from the true origin. So in order to maintain the premise of the test, we need to reconfigure to a relative URL